plaidml/edsl/edsl.h

```diff
@@ -809,7 +813,37 @@ inline Tensor floor(const Tensor& x) { return intrinsic("floor", x); }
 /// \param y Tensor
 /// \return Tensor
 ///
-inline Tensor gather(const Tensor& x, const Tensor& y) { return intrinsic("gather", x, y); }
+class gather {
+ public:
+  explicit gather(const Tensor& x, const Tensor& y) : x_(x), y_(y) {}
+
+  gather& axis(int axis) {
+    axis_ = Tensor(axis);
+    return *this;
+  }
+
+  gather& mode(InterpolationMode m) {
+    mode_ = Tensor(static_cast<uint64_t>(m));
+    return *this;
+  }
+
+  gather& cubic_coeff(float cubic_coeff) {
+    cubic_coeff_ = Tensor(cubic_coeff);
+    return *this;
+  }
+
+  Tensor build() const {
+    std::vector<Tensor> args = {x_, y_, axis_, mode_, cubic_coeff_};
+    return intrinsicCall("gather", args);
+  }
+
+ private:
+  Tensor x_;
+  Tensor y_;
+  Tensor axis_ = Tensor(0);
+  Tensor mode_ = Tensor(static_cast<uint64_t>(InterpolationMode::linear));
+  Tensor cubic_coeff_ = Tensor(-0.5);
```

Comment:
My previous comment here was a mistake; these are eDSL primitive instead of oplib calls. I apologize. However I'm now wondering if we should use `Value` for eDSL primitive operands instead of `Tensor`.